### PR TITLE
Add icons and reorganise history menu

### PR DESCRIPTION
**Pull request by hexylena**
I can't find a bug so it was probably IRC that I was complaining about this. The history menu is hard to scan and interpret visually, hard to find what you're looking for. This was a huge problem with the admin menu a while back until it got reorganised. Here I've attempted to add icons to the history menu and reorganise it a bit to make it easier for people to quickly pick out the things they're interested in.

I believe it's an improvement over the old one:

gx-history-menugx-history-menu1

I'm not very familiar with the JS side of Galaxy so please let me know if I've implemented any anti-patterns.

### CODE REVIEW

**Comment by martenson** [comment]
I like the idea but I think we should maintain clear meaning for every icon site-wide. I have put this together a while ago in order to help doing that: https://wiki.galaxyproject.org/Develop/Graphics

some notes 
- Both items in Downloads have the same icon (+ we usually use http://fortawesome.github.io/Font-Awesome/icon/floppy-o/ ). Also downloads may be under History subtopic?
- Purge is missing
- Dataset security could be better of with http://fortawesome.github.io/Font-Awesome/icon/lock/
- Delete actions are usually iconized with **X**

The rest I mostly like. I will give it another look tomorrow.

**Comment by guerler** [comment]
We often use the trash icon to indicate delete actions i.e. repeat blocks, files in the file uploader and windows in the Scratchbook. However, I agree that we should double check and maintain overall consistency other than that +1.

**Comment by carlfeberhard** [comment]
So keep in mind that, outside the question of icon usability, I'd like to aim to simplify the menu by _removing options_ to places that make more sense - making the icon question moot, but:

The history uses a lot of icons and I've looked into this question before. It turns out that icons by themselves aren't can decrease usability. Expert users (like us - in the sense that we're familiar with the UI) tend to think they assist in navigation and are concise. They are, however, notoriously difficult to communicate what will happen _accurately_ when a button/link is clicked. According to many it's the **label** (and then the position in the menu) that communicates best to a user:
http://uxmyths.com/post/715009009/myth-icons-enhance-usability
http://ux.stackexchange.com/questions/5084/best-practice-for-navigation-usability

Note that the above states 'icons without labels' are less usable. Although I understand the impetus, in this case I don't think the icons are adding value/usability - the labels are (and the reliable positions). If this was a situation where: 
we had very little real estate, 
we displayed them in togglable form - icons + labels toggling to just icons (like github's right hand tabs: pull requests, issues, etc.), 
or even if we did this consistently across the site in our menus
then, I could get behind this request and move to using some icons.

tldr: -0.7452229 heart's in the right place but icons don't do what we think they do - I'd rather remove options from the menu.

**Comment by hexylena** [comment]
@martenson 

> https://wiki.galaxyproject.org/Develop/Graphics

oh, hey.

> Both items in Downloads have the same icon (+ we usually use http://fortawesome.github.io/Font-Awesome/icon/floppy-o/ ). Also downloads may be under History subtopic?

The proper download icon seemed more sane to me, but can switch to the floppy.

>  Purge is missing

No, it's the darker version of the trash can icon, and it's because user_dataset_purge is false on that server. Given your link on icon standards, I'll switch it to the X

> Dataset security could be better of with http://fortawesome.github.io/Font-Awesome/icon/lock/
> Delete actions are usually iconized with X

thanks!

@carlfeberhard 

> So keep in mind that, outside the question of icon usability, I'd like to aim to simplify the menu by removing options to places that make more sense - making the icon question moot, but

100% agreed, and I've seen excellent progress with the new history switcher. All of that will go a long way to simplifying this menu. 

> Note that the above states 'icons without labels' are less usable. Although I understand the impetus, in this case I don't think the icons are adding value/usability - the labels are (and the reliable positions). 

I still use this menu a lot, and I **still**, after multiple years of using Galaxy, have a hard time _quickly_ finding what I want in this menu. I have an incredibly tough time scanning this menu to quickly identify which options I want. I think this is largely due to the poor organisation/poor label choices (e.g. if all history actions were codified with the word "History" and dataset actions were with "Datasets" that would improve)

You raise a fair point, this may be a negative at its current status-quo. I think by adding icons to everything I've not helped the situation. C.f. http://ux.stackexchange.com/a/35974

> If this was a situation where:
> we had very little real estate,
> we displayed them in togglable form - icons + labels toggling to just icons (like github's right hand tabs: pull requests, issues, etc.),
> or even if we did this consistently across the site in our menus
> then, I could get behind this request and move to using some icons.

Given that we don't have little real estate (and this is counter to your point above that icons w/o labels are unhelpful), and that we aren't displaying them toggleable on some smaller menu, I think we had two different ideas of what this PR was accomplishing. I'm **not** trying to simplify this menu or make it more accessible for power users. I'm just trying to fix the UX problem I've experienced for years where I waste seconds scanning the menu every time I use it.

All:

After replying to @carlfeberhard, I think we should scale back icon usage and re-organise some more. Heck, simply organising into "History Actions" and "Dataset Actions" is helping me significantly. 
- Can y'all think of even more improvements to menu organisation w/r/t menu headings/menu item groupings?
- I'll try and scale back icon usage. (How does one make an equal sized empty box as a fa-icon?)

**Comment by carlfeberhard** [comment]
Apart from the icons, I think the new grouping and headers are really good. 

I think a lot of visual scanning speed could be gained by styling those so that they're more distinct than the entries. As it is, they're not too much lighter than the entries and the small caps doesn't differentiate enough to me. A horizontal rule also seems like a possibility between sections.

It might also be possible to increase scanning speed by reducing the number of words for each entry. It looks to me like the 'history actions' section could be reduced to one word per entry. 'dataset actions' may be more difficult.

I'm not sure you need the 'other actions' - you might be able to combine that and the downloads into a 'Import/Export' section (unless that seems too jargony).

Again, I'd rather not see any icons in the menu, but, I believe you can add a 'fa-square' that uses 'color: transparent'.

**Comment by guerler** [comment]
On a side note, "Extract workflow" might fit better into History actions.

**Comment by hexylena** [comment]
@guerler okay, I'll update this PR monday with everyones' suggestions when I'm back from vacation

**Comment by hexylena** [comment]
Okay, 
- removed the icons per @carlfeberhard's request. (I'm surprisingly okay with it)
- moved extract workflow per @guerler's request.

history

**Comment by martenson** [comment]
:+1:

File: client/galaxy/scripts/mvc/history/options-menu.js [hunk 1]
```diff
@@ -177,7 +191,7 @@ var create = function( $button, options ){
         purgeAllowed = options.purgeAllowed || false,
         root = options.root || ( ( Galaxy && Galaxy.options )? Galaxy.options.root: '/' ),
         menu = buildMenu( isAnon, purgeAllowed, root );
-    //console.debug( 'menu:', menu );
+    console.debug( 'menu:', menu );
```
Comment (nsoranzo) [inline client/galaxy/scripts/mvc/history/options-menu.js]: Should this stay commented?

Comment (hexylena) [inline client/galaxy/scripts/mvc/history/options-menu.js]: Yes

Comment (jmchilton) [inline client/galaxy/scripts/mvc/history/options-menu.js]: @erasche please re-comment and I'll :+1:.

**Comment by carlfeberhard** [comment]
+1

**Comment by hexylena** [comment]
Fixing debug line now.

**Comment by hexylena** [comment]
Ok, Done.

**Comment by martenson** [comment]
thank you @erasche !
